### PR TITLE
String.new(capacity: size)

### PR DESCRIPTION
**Pull request by andrykonchin**
Related issue https://github.com/ruby/spec/issues/473

### CODE REVIEW

**Comment by andrykonchin** [comment]
@eregon I see a lot of `ruby_version_is ''...'2.3' do' and `ruby_version_is '2.3' do` version guards.
Does it mean these specs don't run for Ruby 2.4? Should these guards be updated?

**Comment by eregon** [comment]
`ruby_version_is '2.3' do` means RUBY_VERSION >= 2.3. so they apply on 2.4 and future version too.
`ruby_version_is ''...'2.3'` means RUBY_VERSION < 2.3. It's a bit confusing and we need better documentation, sorry.

File: core/string/new_spec.rb [hunk 1]
```diff
@@ -15,6 +15,12 @@
     end
   end
 
+  ruby_version_is "2.4" do
+    it "accepts a capacity argument" do
+      lambda { String.new("", capacity: 100_000) }.should_not raise_error
```
Comment (eregon) [inline core/string/new_spec.rb]: Indeed, it's hard to know if this did anything.
`should_not raise_error` is the least useful matcher, could you replace it with `.should == ""` and
add an example where the first String argument is not empty?

Comment (andrykonchin) [reply core/string/new_spec.rb]: @eregon OK, done

**Pull request merged**